### PR TITLE
File Watcher 

### PR DESCRIPTION
**Pull request by barelyhuman**
Implement a basic file watcher and live reload

### CODE REVIEW

**Comment by Aslemammad** [comment]
Shall I do anything on this?

File: src/watcher.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  dependsOn: ['nomen:builder'],
+  async onLoad(ctx) {
+    if (!ctx.env.NOMEN_DEV) return
+
+    const watcherInstance = chokidar.watch('.', {
+      ignored: [/.nomen/, /node_modules/],
+      awaitWriteFinish: true,
+    })
+
+    watcherInstance.add('./**/*.js')
```
Comment (Aslemammad) [inline src/watcher.js]: this expects all files to be js?

**Comment by barelyhuman** [comment]
> Shall I do anything on this?

Nah, this is almost done, I just need to add it to the preact and arrowjs modules as well. live reload seems to be working with vanilla and then you can use the same socket module to inject HMR related stuff as well

File: src/watcher.js (hunk 1 above)
Comment (barelyhuman) [reply src/watcher.js]: that's initial implementation code. 

The final version would be `./**/*.{js,mjs,ts,tsx,jsx}`

**Pull request merged**